src/downloader.rs

```diff
@@ -68,8 +70,12 @@ impl Downloader {
             auth_map.insert(host, (username, password));
         }
 
+        let mut headers = HeaderMap::new();
+        headers.insert(COOKIE, HeaderValue::from_str(cookie).unwrap());
```

Comment:
I fear this will not work over multiple domains, as you would always give the same cookies without taking into consideration the domains. What if 2 domains want a cookie named `test`?

If you have time (and want to go further with this), have a look there:
- https://docs.rs/reqwest/latest/reqwest/cookie/struct.Jar.html#method.add_cookie_str
- https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.cookie_provider

In any case this is better than what we have now so I will take this, with the modifications or not. Let me know